src/components/loginRegister/registerRequest.js

```diff
@@ -12,7 +12,10 @@ export const register = async (email, password) => {
     });
 
     if (!response.ok) {
-      throw new Error("Registration failed");
+      throw {
+        message: "Registration failed",
+        status: response.status,
+      };
```

Comment:
Lets only throw Error objects.
For example like this:
`throw new Error('Registration failed')`

Thats the convention for javascript developers for the most part. Using an Error object rather than just an object provides a stack trace too which can be useful sometimes for finding where an error started.
This is what a stack trace is:
![js-stack-trace](https://github.com/rknizzle/propermesh-view/assets/40612461/4991c638-646f-44ae-8beb-615bdca8dd9e)

Reply:
I am going to have to familiarize myself with the ins and outs of a stack trace. 

I see them all the time but never know how to read them

Reply:
Yea theres usually a bunch of gibberish in them so they're hard to figure out. Usually at the top or bottom though they'll kind of show you which file the error originated from